tangelo/toolboxes/molecular_computation/integral_solver_pyscf.py

```diff
@@ -108,14 +111,21 @@ def compute_mean_field(self, sqmol):
         sqmol.mean_field = self.scf.RHF(molecule) if not sqmol.uhf else self.scf.UHF(molecule)
         sqmol.mean_field.verbose = 0
 
+        if self.chkfile is not None:
+            is_chkfile_exist = os.path.exists(self.chkfile)
+            sqmol.mean_field.chkfile = self.chkfile
+        else:
+            is_chkfile_exist = False
+
         # Force broken symmetry for uhf calculation when spin is 0 as shown in
         # https://github.com/sunqm/pyscf/blob/master/examples/scf/32-break_spin_symm.py
-        if sqmol.uhf and sqmol.spin == 0:
+        if sqmol.uhf and sqmol.spin == 0 and not is_chkfile_exist:
             dm_alpha, dm_beta = sqmol.mean_field.get_init_guess()
             dm_beta[:1, :] = 0
             dm = (dm_alpha, dm_beta)
             sqmol.mean_field.kernel(dm)
         else:
+            sqmol.mean_field.init_guess = "chkfile" if is_chkfile_exist else "minao"
```

Comment:
So, if a user provides a path to a check file that is incorrect or does not exist, you're not going to tell them or raise an error, instead you're gonna do computations and make them believe everything went according to their plan ?

I'd be tempted to raise a warning or an error as soon as I realize the check file cannot be found. Do you think computing from scratch if it cannot be found is desirable behaviour for users who explicitly tried to provide a chkfile ? In what cases ?

Reply:
If there is no chkfile already existing, the code will create one (and the initial guess stays the default one). Without that, the users wouldn't be able to create a chkfile, preventing them to use the feature of getting initial guesses from this file for future calculations. It is actually the expected behaviour.